### DISCUSSION TITLE
Any way to check which timers are running ?

### DISCUSSION THREAD

**Discussion by shettyh** [Kong/kong#10861] [Help] [answered]
Hi,

In one of the environments we see lot of pending nginx timers ( 10x of running timers ), is there any way to check what are these timers that are running ?

**Comment by hanshuebner** [comment, marked as answer]
Hi @shettyh,

the [timer-ng](https://github.com/Kong/lua-resty-timer-ng#stats) library (used in Kong 3.x) provides a [`stats` function](https://github.com/Kong/lua-resty-timer-ng#stats) that you can invoke to retrieve information about all timers that are currently defined.  The function is not exposed as an API call, but you could use a custom plugin to make it available.

Does that help?

Kind regards,
Hans

**Reply by shettyh** [reply]
@hanshuebner thanks, that helps. But i see this library is used only from 3.x, not able to see in 2.8.3 


Also seeing p99 timer delay spikes more than 100ms, what could be the reason for this ?

**Reply by hanshuebner** [reply]
> i see this library is used only from 3.x, not able to see in 2.8.3

You're right - We've ported the timer-ng changes to the 2.8.x branch but did not release from that.

> Also seeing p99 timer delay spikes more than 100ms, what could be the reason for this ?

Is this with 2.8.3 or 3.x?

**Reply by shettyh** [reply]
This is in 2.8.3

**Reply by hanshuebner** [reply]
@shettyh At this point, it would be best if you could upgrade to 3.x, as the new timer library certainly has a completely different behavior and may solve your problem.

**Reply by shettyh** [reply]
Sure thanks, will update the status post upgrade
